datetime/datetime.go

```diff
@@ -109,6 +109,7 @@ func NewDatetime(t time.Time) (*Datetime, error) {
 			return nil, fmt.Errorf("unknown timezone %s with offset %d",
 				zone, offset)
 		}
+		t = t.In(time.FixedZone(zone, offset))
```

Comment:
Is it a go in-built `time` -> go-tarantool `datetime` conversion?

Reply:
Yeah, we convert a time with a  Location to a time with a fixed timezone here.

Reply:
So we are losing full information? What's the reason? It it impossible to work with non-fixed timezones?

Reply:
1. A user creates a time with a default Location `Europe/Moscow`.
2. We return timestamp to a user with `Europe/Moscow`.
2. We encode it as `MSK`.
4. We decode it as `MSK`.
5. We return timestamp to a user with `MSK`.

It's better always return a time with `MSK` to the user. Because it will work the same way with the time change.

Reply:
> We encode it as MSK.
Why?

Reply:
> It's better always return a time with MSK to the user.

Both approaches are confusing for a user.

Reply:
> Both approaches are confusing for a user.

So we need to choose to make it work in the same way for all cases =)

We can't create a custom implementation of `Location` in Golang, so we have only two options:

1. FixedZone
2. Default locations.

FixedZone will be easy to support:

1. Default locations may change it value: `Europe/Moscow` may be `MSK` or `MSD`.
2. Default locations in GoLang depend on a platform configuration, see: https://pkg.go.dev/time#LoadLocation

Reply:
> We can't create a custom implementation of Location in Golang

Why we can't use the default one? If the answer is

> Default locations in GoLang depend on a platform configuration, see: https://pkg.go.dev/time#LoadLocation

I don't get it: if we don't have any location info, we wouldn't be able to convert it to some FixedZone too.

So how do we work with datetime if Tarantool sends `Europe/Moscow`? Convert it to `MSK`? They are not equivalent, this is wrong.

> Default locations may change it value: Europe/Moscow may be MSK or MSD
Yeah, because this is how `Europe/Moscow` timezone works. What's the problem? We can't work with locations, only zones?

Reply:
If we can't work with `Europe/Moscow` timezones it is better to raise an error rather that work with them incorrectly.

Reply:
> I don't get it: if we don't have any location info, we wouldn't be able to convert it to some FixedZone too.

Then we will have a different behavior of the public API on different platforms.

> So how do we work with datetime if Tarantool sends `Europe/Moscow`? Convert it to `MSK`? They are not equivalent, this is wrong.

We don't convert `Europe/Moscow` to `MSK` itself. We just use `zone` information about a time value. Why is it wrong?
 
> Yeah, because this is how `Europe/Moscow` timezone works. What's the problem? We can't work with locations, only zones?

We can, but it will break a user code in random places by `Local` location.

Reply:
> If we can't work with `Europe/Moscow` timezones it is better to raise an error rather that work with them incorrectly.

We can work with it:

https://github.com/tarantool/go-tarantool/blob/oleg-jukovec/datetime-non-fixed/datetime/example_test.go#L168

Reply:
> We don't convert `Europe/Moscow` to `MSK` itself. We just use `zone` information about a time value. Why is it wrong?

Because of `tarantool 'Europe/Moscow'` -> `go 'MSK'` -> `tarantool 'MSK' pipeline`

Reply:
> Then we will have a different behavior of the public API on different platforms.

Same as in SSL, by the way.

I don't think it is wrong to assume that Olson database implementation is platform-independent.

Reply:
> We can, but it will break a user code in random places by `Local` location.

I'm not well-experienced in Go `time`, could you explain this?

Reply:
> > We don't convert `Europe/Moscow` to `MSK` itself. We just use `zone` information about a time value. Why is it wrong?
> 
> Because of `tarantool 'Europe/Moscow'` -> `go 'MSK'` -> `tarantool 'MSK'` pipeline

That's why we use FixedZone:
`tarantool 'Europe/Moscow' -> `go 'Europe/Moscow'` -> `tarantool 'Europe/Moscow`

Reply:
> > We can, but it will break a user code in random places by `Local` location.
> 
> I'm not well-experienced in Go `time`, could you explain this?

As example:
```Go
tm, err := time.Parse(time.RFC3339 + " MST", "2006-01-02T15:04:00+03:00 MSK")
// tm.Location.String() == "Local" if you are in "MSK" location
// tm.Location.String() == "MSK" if you are not in "MSK" location
```

Reply:
> > Then we will have a different behavior of the public API on different platforms.
> 
> Same as in SSL, by the way.

Not quite. In SSL case, the code either works or it doesn't. In the second case, the public API works differently.

Reply:
Now I would prefer to create a separate class with the constructor: (year, month...timezone string) like we act in Python.

But now we need to support correct and non-dual behavior in the current code.